routes/file.js

```diff
@@ -4,12 +4,25 @@ module.exports = async ( request, reply ) => {
 	const mime = require( 'mime-types' );
 	const minify = require( '../lib/minify' );
 	const compress = require( '../lib/compress' );
+	const validatePath = require( '../lib/validate-path' );
 	const { envBool } = require( '../lib/util' );
 	const { PerformanceObserver, performance } = require( 'perf_hooks' );
 
-	const path = request.query.path;
-
 	let log = {};
+
+	const userPath = request.query.path;
+
+	// Check user's path falls in the base path.
+	const path = validatePath( userPath );
+	if ( path === false ) {
+		// Error should look the same as a stat failed, in order to
+		// not leak information
+		const err = new Error( `ENOENT: no such file or directory, open '${ userPath }'` );
```

Comment:
My first instinct would be to replace the system error with a custom message (in the legitimate 404 scenario) by using a try/catch with a throw in the catch block, rather than attempting to duplicate the system message in the outside-the-base-path scenario. That seems safer in the long run, and less likely to leak should anything change.

I realise that it's effectively an API change, but I think we're early enough in the lifetime of the minifier that no-one is relying on specifically-worded error messages.